web/src/auth/CasLogout.js

```diff
@@ -35,24 +35,53 @@ class CasLogout extends React.Component {
   UNSAFE_componentWillMount() {
     const params = new URLSearchParams(this.props.location.search);
 
+    const logoutInterval = 100;
+    const logoutTimeOut = (s, redirectUri, linkSoftFlag) => {
+      setTimeout(() => {
+        AuthBackend.getAccount().then((accountRes) => {
+          if (accountRes.status === "ok") {
+            AuthBackend.logout().then((logoutRes) => {
+              if (logoutRes.status === "ok") {
+                const redirectData = logoutRes.data2;
+                if (redirectData !== null && redirectData !== undefined && redirectData !== "") {
+                  logoutTimeOut(s, redirectData, false);
+                } else if (params.has("service")) {
+                  logoutTimeOut(s, params.get("service"), false);
+                } else {
+                  logoutTimeOut(s, `/cas/${this.state.owner}/${this.state.applicationName}/login`, true);
+                }
+              } else {
+                Setting.showMessage("error", `Failed to log out: ${logoutRes.msg}`);
+              }
+            });
+          } else {
+            if (linkSoftFlag) {
+              Setting.goToLinkSoft(this, redirectUri);
+            } else {
+              Setting.goToLink(redirectUri);
+            }
+            Setting.showMessage("success", "Logged out successfully");
+            this.props.onUpdateAccount(null);
+          }
+        });
+      }, s);
```

Comment:
you pass s to func `logoutTimeOut`, why not replace s with `logoutInterval` ?

Reply:
ok, I used a variable name that is not easy to understand and I will correct it in a moment.